src/plugins/controls/public/control_types/options_list/options_list_editor.tsx

```diff
@@ -122,6 +167,16 @@ export const OptionsListEditor = ({
           }}
         />
       </EuiFormRow>
+      <EuiFormRow>
+        <EuiSwitch
+          label={OptionsListStrings.editor.getRunPastTimeoutTitle()}
+          checked={Boolean(state.runPastTimeout)}
+          onChange={() => {
+            onChange({ runPastTimeout: !state.runPastTimeout });
+            setState((s) => ({ ...s, runPastTimeout: !s.runPastTimeout }));
```

Comment:
This is actually because the controls are set up to use a _different_ timeout. This is because speed matters more in autocomplete / typeahead use cases, so it is common to want a lower timeout in the autocomplete settings. This is actually controlled by `data.autocomplete.valueSuggestions.timeout`and `data.autocomplete.valueSuggestions.terminateAfter` which can be set in the `kibana.yml`. Much less user friendly unfortunately.

I will double check this behaviour, but I'm pretty sure it will work properly because when `runPastTimeout` is on the `timeout` and `terminate_after` settings simply aren't sent along with the query, which makes ES treat the query like a normal terms query, and run until it fetches _everything_.

Reply:
Ahhh okay okay, that makes a lot more sense! Thanks for clarifying 👍 Definitely think we need some copy clarification here, since I don't think it's obvious how this timeout works and/or what it is used for. Plus maybe some info on how to set it? But that might be too much 😆 @KOTungseth